### PR TITLE
OBSDATA-492: Add confluentinc specific changes to master

### PR DESCRIPTION
**Pull request by CodingParsley**
### Description
https://confluentinc.atlassian.net/browse/OBSDATA-492
Add confluentinc specific changes to master
Changelog:

* Adding minikube-setup instructions.

Update tiny-cluster.yaml to make it work in minikube.

Fixing operator startup bug.

Miscellaneous fixes to make local minikube setup work.

Fixing MM readiness probe and steps for minikube ingress-dns issue for local minikube setup.

* semaphore (#5)

* Add command args to container creation (#6)

* go/codeowners: Generate CODEOWNERS [ci skip] (#7)

* [METRICS-4348] update obs-data team as codeownderswq (#8)

* [METRICS-4487] add obs-oncall as codeowners (#10)

* DP-8085 - Migrate to Sempahore self-hosted agent (#9)

* DP-9370 - Migrate to Semaphore self-hosted agent (#15)

* chore: update repo semaphore project

* Update service.yml (#17)

Update semaphore job commands and go version

update build triggers

update semaphore whitelist

Update project.yml

* update CI 

* nodespec should take precedence (#13)

<hr>

This PR has:
- [x] been tested on a real K8S cluster to ensure creation of a brand new Druid cluster works.(tested in stag)
- [x] been tested for backward compatibility on a real K*S cluster by applying the changes introduced here on an existing Druid cluster. If there are any backward incompatible changes then they have been noted in the PR description.
- [x] added comments explaining the "why" and the intent of the code wherever would not be obvious for an unfamiliar reader.
- [x] added documentation for new or modified features or behaviors.

<hr>

[METRICS-4348]: https://confluentinc.atlassian.net/browse/METRICS-4348?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: .semaphore/project.yml [hunk 1]
```diff
@@ -0,0 +1,37 @@
+apiVersion: v1alpha
```
Comment (hampycapper) [inline .semaphore/project.yml:1]: Are you merging these from upstream? Or are these things we've added?

**Review by hampycapper — COMMENTED** [review]
If these are all merged from upstream, we can move forward...

**Comment by CodingParsley** [comment]
@hampycapper ready for another look:)

File: Makefile [hunk 2]
```diff
@@ -16,7 +16,7 @@ SHELL = /usr/bin/env bash -o pipefail
 .SHELLFLAGS = -ec
 
 .PHONY: all
-all: build
+all: build test lint template docker-build
```
Comment (hampycapper) [inline Makefile:19]: Do you need this, since you have semaphore doing the same above? 

Meaning, if I type "make" in my own dev machine, should I be doing all that stuff?

File: examples/tiny-cluster-zk.yaml [hunk 3]
```diff
@@ -53,11 +53,8 @@ spec:
             - containerPort: 3888
               name: zk-elec-port
           resources:
-            limits:
-              cpu: 1
-              memory: 512Mi
             requests:
-              cpu: 1
+              cpu: 100m
```
Comment (hampycapper) [inline examples/tiny-cluster-zk.yaml:57]: Why this change for tiny-cluster?

File: examples/tiny-cluster.yaml [hunk 4]
```diff
@@ -6,7 +6,7 @@ kind: "Druid"
 metadata:
   name: tiny-cluster
 spec:
-  image: apache/druid:0.22.1
+  image: confluent-docker.jfrog.io/confluentinc/cc-druid:v1.202.0
```
Comment (hampycapper) [inline examples/tiny-cluster.yaml:9]: Didn't we move away from jfrog?

File: examples/tiny-cluster.yaml [hunk 5]
```diff
@@ -52,26 +52,46 @@ spec:
...
     druid.zk.paths.base=/druid
     druid.zk.service.compress=false
 
     # Metadata Store
-    druid.metadata.storage.type=derby
-    druid.metadata.storage.connector.connectURI=jdbc:derby://localhost:1527/druid/data/derbydb/metadata.db;create=true
-    druid.metadata.storage.connector.host=localhost
-    druid.metadata.storage.connector.port=1527
-    druid.metadata.storage.connector.createTables=true
+    # druid.metadata.storage.type=derby
```
Comment (hampycapper) [inline examples/tiny-cluster.yaml:64]: Why all these comments? Why. not just make the change and rely on repo history?

**Review by hampycapper — APPROVED** [review]
This is just a merge, so all changes already landed in other branch.
